client/src/components/gameBoardComponent/gameBoardComponent.js

```diff
@@ -0,0 +1,107 @@
+import React, { Component } from 'react';
+import { Redirect } from 'react-router-dom';
+import PropTypes from 'prop-types';
+
+import backgroundImage from '../../static/images/sand.jpg';
+import './gameBoardComponent.css';
+import GAME_CONSTANTS from './gameConstants';
+
+export default class GameBoardComponent extends Component {
+  constructor(props) {
+    super(props);
+    const { playerInfo } = props.location;
+    if (playerInfo) {
+      const { playerId, socket } = playerInfo;
+      this.state = {
+        playerId,
+      };
+      this.leaveGame = this.leaveGame.bind(this);
+      this.socket = socket;
+    } else {
+      this.redirectToHome = true;
+    }
+  }
+
+  componentDidMount() {
+    if (!this.redirectToHome) {
+      this.img = document.getElementById('backgroundImage');
+      this.ctx = this.canvas.getContext('2d');
+      this.socket.on('stepChange', (game) => {
+        this.stepChange(game);
+      });
+    }
+  }
+
+
+  handleKeyDown = (event) => {
+    const keyNumber = event.keyCode;
+    const { key } = event;
+    // we only send the event if the event is right, left, top or bottom arrow
+    // these integers are numeric value of these arrows on keyboard.
+    if ([37, 38, 39, 40].find(number => number === keyNumber)) {
```

Comment:
I'd say don't use keycodes. From the [documentation](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode):

> "This feature is no longer recommended. Though some browsers might still support it, it may have already been removed from the relevant web standards, may be in the process of being dropped, or may only be kept for compatibility purposes."